manager/src/main/java/com/redhat/service/smartevents/manager/SecretsServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+package com.redhat.service.smartevents.manager;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.ProcessorLifecycleException;
+import com.redhat.service.smartevents.infra.models.EventBridgeSecret;
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+import com.redhat.service.smartevents.infra.models.processors.ProcessorDefinition;
+import com.redhat.service.smartevents.manager.models.Processor;
+import com.redhat.service.smartevents.manager.providers.ResourceNamesProvider;
+import com.redhat.service.smartevents.manager.vault.VaultService;
+import com.redhat.service.smartevents.processor.ProcessorCatalogService;
+
+import io.smallrye.mutiny.Uni;
+
+import static com.redhat.service.smartevents.manager.SecretsService.emptyObjectNode;
+import static com.redhat.service.smartevents.manager.SecretsService.mergeObjectNodes;
+
+@ApplicationScoped
+public class SecretsServiceImpl implements SecretsService {
+
+    @ConfigProperty(name = "secretsmanager.timeout-seconds")
+    int secretsManagerTimeout;
+
+    @Inject
+    ObjectMapper mapper;
+    @Inject
+    ProcessorCatalogService processorCatalogService;
+    @Inject
+    ResourceNamesProvider resourceNamesProvider;
+    @Inject
+    VaultService vaultService;
+
+    @Override
+    public void maskProcessor(Processor processor) {
+        // deep copy existing definition since maskProcessorDefinition directly updates the received one
+        ProcessorDefinition maskedDefinition = processor.getDefinition().deepCopy();
+        Map<String, ObjectNode> newSecrets = maskProcessorDefinition(maskedDefinition);
+        boolean newHasSecrets = !newSecrets.isEmpty();
+
+        if (processor.hasSecrets() && !newHasSecrets) {
+            // processor used to have secrets and now has none, so we must delete the existing secrets
+            await(vaultService.delete(secretNameFor(processor)));
+        } else if (newHasSecrets) {
+            // otherwise we must create/update the existing secrets
+            Map<String, String> serializedNewSecrets = serializeSecrets(newSecrets);
+            EventBridgeSecret eventBridgeSecret = new EventBridgeSecret(secretNameFor(processor), serializedNewSecrets);
+            vaultService.createOrReplace(eventBridgeSecret).await().atMost(Duration.ofSeconds(secretsManagerTimeout));
+        }
+
+        // update processor
+        processor.setDefinition(maskedDefinition);
```

Comment:
It's probably just my opinion; but if we're cloning part of a `Processor` should we not really clone it all?

It would have been beneficial to RHOSE (or any multi-threaded, distributed system) if we didn't mutate *any* object and always returned new instances.. but that moment has long passed. I just feel having a partial-mutated, partial-cloned object could be dangerous in the future (dangerous in the sense that future developers will have to understand the behaviour of the class).